### ISSUE TITLE
Fasten in cpu

### ISSUE THREAD

**Issue by Zumbalamambo** [timctho/ConvolutionalPoseMachines-Tensorflow#1]
It is laggy in cpu. how can I optimize it?

**Comment by timctho** [comment]
You might need a Nvidia GPU that supports CUDA to run in realtime. For CPU, check if Tensorflow provides any optimization for it. And of course you can reduce the input image size or train a smaller model by yourself to get a speedup.

**Comment by Zumbalamambo** [comment]
Can you please tell me how I can train my own model?

**Comment by justinluyao** [comment]
hi !

thanks for your work！

can you tell me what kind of data you used for training?

**Comment by timctho** [comment]
@Zumbalamambo 
You need to feed placeholder with "input image", "center map", "joint heat map", please read the paper for more detail. I will release my training script after some refactors.

@justinluyao 
I manually labeled thousands of images with 2D joint locations, and generate some 3D model images from LibHand. If you find publicly available dataset please let me know too.

**Comment by justinluyao** [comment]
do you mean you generate the hand images with LibHand and label the joints positions on the generated images by yourself?

**Comment by timctho** [comment]
LibHand can not only generate images but also with joints labels, by projecting 3D model joints onto 2D image spaces.

**Comment by sarahzhouUestc** [comment]
@timctho Hi timctho, thanking for you sharing of your project. I am doing my training with it. However I have some confusion about it, could you please do me a favor?

1. Need I feed background channel? cause I noticed that shihenw uses 4th channel which represents the background, and its values are all 0. actually I don't know why need to do this.  While in your project, the input image still have 3 channels, right? and I just need to generate background heatmap, then the total number of groundtruth heatmap is (nun_of_joints + 1), am i right?
2. The center of center_map is the center of torsobox, right?  but what is the sigma of it , how can I determine it?

Thank you very much!

**Comment by timctho** [comment]
1.For original CPM paper I think the input image only have rgb channels.
And yes you need to generate the background heatmap which is complement to
all the joint heatmaps.
2.Yes. In hand project, I remember the sigma is 7 or nearby. You can play
with it a little, while I find out little difference.

Best wishes,
*Tim Ho* (何哲廷)
